### PR TITLE
cadical cdclt: Some fixes for incremental CaDiCaL(T).

### PR DESCRIPTION
**Pull request by mpreiner**
- Add activation literals to explanations
- Fix value of popped activation literal -> huge performance impact
- Do not observe activation literals -> reduces notification overhead
- Only perform theory propagations when all activation literals have been assigned
- Refactors addClause() to always pass clause to propagator if configured.

### CODE REVIEW

**Comment by mpreiner** [comment]
Cluster run results follow.

**Comment by mpreiner** [comment]
Changes 2) and 3) from above make the performance difference compared to the main version, still not there, but working on it.
```
                      total   solved      sat    unsat  time_cpu
config
cvc5-minisat-inc      43287  5317080  4140638  1176442 2657552.1
cvc5-cadical-inc      43287  4694921  3596291  1098630 2442930.0
cvc5-cadical-main-inc 43287  2609713  1937475   672238 2701750.6
```

File: src/prop/cadical.cpp [hunk 1]
```diff
@@ -524,8 +540,20 @@ class CadicalPropagator : public CaDiCaL::ExternalPropagator
       {
         lits.insert(lits.begin(), toCadicalLit(alit));
       }
-      d_new_clauses.insert(d_new_clauses.end(), lits.begin(), lits.end());
-      d_new_clauses.push_back(0);
+      // Buffer clause
```
Comment (aniemetz) [inline src/prop/cadical.cpp:543]: What is the purpose of this clause? Document

**Review by aniemetz — APPROVED** [review]
One minor thing, else LGTM.

**Pull request merged**